ES6 Katas - iterator

shinenelson opened the pull request:
~I wasn't able to solve the last kata #40 - usages~

~I have a feeling that there' i's something wrong with the tests itself.~
~I've reached out to someone who knows better than me to validate it for me.~

shinenelson committed:
[es6][iterator] array

shinenelson committed:
[es6][iterator] string

shinenelson committed:
[es6][iterator] protocol

shinenelson committed:
[es6][iterator] usages

finally! after 3.5 hours of racking my brain
not counting the 2+ hours I spent on the same
problem on a previous day

sirajulm commented on language/es6/iterator/usages.js:
```diff
@@ -29,11 +29,11 @@ describe('Iterator usages', () => {
       }
     }
-    usersIterable = {};
+    usersIterable = { [Symbol.iterator]: () => iteratorFunction() };
```
```suggestion
    usersIterable = { [Symbol.iterator]: iteratorFunction };
```
wrapper function not necessery.

shinenelson committed:
[es6][iterator] remove function wrapper

resolves review [comment #1](https://github.com/shinenelson/learning-ecmascript/pull/13#discussion_r434549969) by @sirajulm
